src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -108,6 +108,14 @@ public TracerSettings(IConfigurationSource source)
             GlobalTags = GlobalTags.Where(kvp => !string.IsNullOrEmpty(kvp.Key) && !string.IsNullOrEmpty(kvp.Value))
                                    .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
 
+            HeaderTags = source?.GetDictionary(ConfigurationKeys.HeaderTags) ??
+                         // default value (empty)
+                         new ConcurrentDictionary<string, string>();
```

Comment:
Do we need `ConcurrentDictionary`? `Dictionary` can be safely read from multiple threads. See `GlobalTags`.

Reply:
Our implementations by default (including `JsonConfigurationSource.GetDictionary` and `StringConfigurationSource.ParseCustomKeyValues` return `ConcurrentDictionary`. Should we consider changing the default for all of these `Dictionary`?

Reply:
This is quickly getting out of scope for this PR. This is good for now and all the `ConcurrentDictionary` work. Maybe @kevingosse will want to clean them up later if it's a performance concern.